_drafts/practical_zephyr_02_dt_basics.md

```diff
@@ -0,0 +1,1468 @@
+
+> "A property value is an **array** of zero or more bytes that contain information associated with the property. Properties might have an empty value if conveying true-false information. In this case, the presence or absence of the property is sufficiently descriptive." [[DTSpec]](https://www.devicetree.org/specifications/)
+
+This is basically just a fancy way of saying that devicetree supports properties without a value assignment, as well as properties with one or multiple values. The [[DTSpec]](https://www.devicetree.org/specifications/) then goes ahead and provides the following possible _"property values"_ and their representation in memory. In our table, we can skip the memory representation since we've learned that [Zephyr doesn't compile the DTS files into a binary](#compiling-the-devicetree):
+
+| Type                   | Description                                                            |
+| :--------------------- | :--------------------------------------------------------------------- |
+| `<empty>`              | Empty value. `true` if the property is present, otherwise `false`.     |
+| `<u32>`                | A 32-bit integer in big endian format (a "cell").                      |
+| `<u64>`                | A 64-bit integer in big endian format. Consists of two `<u32>` values. |
+| `<string>`             | A printable, null-terminated string.                                   |
+| `<prop-encoded-array>` | An array of values, where the type is defined by the property.         |
+| `<phandle>`            | A `<u32>` value used to reference another node in the devicetree.      |
+| `<stringlist>`         | A list of concatenated `<string>` values.                              |
+
+> **Note:** The term **"cell"** is typically used to refer to the individual data elements within properties and can thus be thought of as a single unit of data in a property. The [devicetree specification](https://www.devicetree.org/specifications/) v0.4 defines a **"cell"** as _"a unit of information consisting of 32 bits"_ and thus explicitly defines its size as 32 bits. Meaning it is just another confusing name for a 32-bit integer.
+
+If you find this information underwhelming - you're not alone. It also won't be of much help when trying to understand property values from Zephyr's devicetree source files. The missing link to understanding the nature of the above table is the following: Values in Zephyr's DTS files are **represented** using the _"Devicetree Source (DTS) Format"_ version _1_, meaning they use a specific format or **syntax** to represent the above property value types. The above table, on the other hand, is a format-agnostic list of types that any specific DTS value format must translate to. E.g., a new _"Devicetree Source (DTS) Format"_ version _2_ might use a different syntax to represent values of the listed types.
```

Comment:
Yeah, makes sense. I was just jumping back and forth between the sections in the spec. since I'm a bit too much in love with details, but it is very impractical. The only info that is very important from those paragraphs is the note about "cells" (I really don't like that term, I find it very useless :D). I'll think of something.